federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProvider.java

```diff
@@ -459,7 +459,12 @@ protected List<LDAPObject> searchLDAP(RealmModel realm, Map<String, String> attr
                 ldapQuery.addWhereCondition(emailCondition);
 
                 List<LDAPObject> ldapObjects = ldapQuery.getResultList();
-                results.addAll(ldapObjects);
+                ldapObjects.forEach(ldapObject -> {
+                    // ensure that no entity is listed twice and still preserve the order of returned entities
+                    if (!results.contains(ldapObject)) {
+                        results.addAll(ldapObjects);
```

Comment:
```suggestion
                        results.add(ldapObject);
```
(same below)

Reply:
I am worried of the complexity. Could we instead use a set of unique identifiers to check if the `ldapObject` has already been added?

Reply:
@hmlnarik - reworked. Still, if we would have computational complexity here, I assume the result would be too big for the caller anyway / would take too long to retrieve from LDAP.

Reply:
Thank you, resolving this